src/StackExchange.Redis/ChannelMessageQueue.cs

```diff
@@ -162,17 +175,29 @@ private async Task OnMessageSyncImpl()
             while (!Completion.IsCompleted)
             {
                 ChannelMessage next;
-                try { if (!TryRead(out next)) next = await ReadAsync().ConfigureAwait(false); }
-                catch (ChannelClosedException) { break; } // expected
-                catch (Exception ex)
+
+                // Keep trying to read values
```

Comment:
This is the pattern we generally use to read from a channel and gracefully stop without an exception. If there's a reason for the older behavior, I'm happy to reset it.